app/src/main/java/org/mozilla/fenix/library/downloads/DownloadAdapter.kt

```diff
@@ -27,14 +29,38 @@ class DownloadAdapter(
 
     fun updateMode(mode: DownloadFragmentState.Mode) {
         this.mode = mode
+        // Update the delete button alpha that the first item holds
+        if (itemCount > 0) notifyItemChanged(0)
     }
 
     override fun onBindViewHolder(holder: DownloadsListItemViewHolder, position: Int) {
-        holder.bind(downloads[position])
+        val current = downloads[position]
+        val isPendingDeletion = pendingDeletionIds.contains(current.id)
+        holder.bind(downloads[position], position == 0, mode, isPendingDeletion)
     }
 
     fun updateDownloads(downloads: List<DownloadItem>) {
         this.downloads = downloads
         notifyDataSetChanged()
     }
+
+    fun updatePendingDeletionIds(pendingDeletionIds: Set<String>) {
+        this.pendingDeletionIds = pendingDeletionIds
+    }
```

Comment:
Could this function be replaced with `DownloadFragmentAction.AddPendingDeletionSet` ?

Reply:
Answered during our discussion with Jeff